django_x509/base/models.py

```diff
@@ -142,7 +143,12 @@ def clean_fields(self, *args, **kwargs):
         # must be done here in order to validate imported fields
         # and fill private and public key before validation fails
         if not self.pk and self.certificate and self.private_key:
-            self._import()
+            try:
+                crypto.load_certificate(crypto.FILETYPE_PEM,self.certificate)
+                crypto.load_privatekey(crypto.FILETYPE_PEM, self.private_key)
```

Comment:
Ok, I understood what is happenig, sorry. the best thing to do in this case is the following:

- put `self._import()` back where it was
- create a new private method called `_validate_pem`
- move this logic in `_validate_pem` and call it just before `_import`
- please repeat the try except block twice, first for `self.certificate` and then for `self.private_key`, and when you raise the `ValidationError`, do it like this:

`raise ValidationError({'certificate': _('OpenSSL error: {0}'.format(e.args[0]))})`

and:

`raise ValidationError({'private_key': _('OpenSSL error: {0}'.format(e.args[0]))})`

This will tell django to which field the validation error must be added to.